test/suite/test_chunkcache04.py

```diff
@@ -0,0 +1,141 @@
+    
+    pinned_uris = ["table:chunkcache01", "table:chunkcache02"]
+
+    scenarios = make_scenarios(format_values, cache_types)
+
+    def conn_config(self):
+        if not os.path.exists('bucket2'):
+            os.mkdir('bucket2')
+
+        return 'tiered_storage=(auth_token=Secret,bucket=bucket2,bucket_prefix=pfx_,name=dir_store),' \
+            'chunk_cache=[enabled=true,chunk_size=512KB,capacity=20GB,pinned=' \
+                + '("' + '{}'.format("\",\"".join(self.pinned_uris)) \
+                + '"),type={},storage_path=WiredTigerChunkCache]'.format(self.chunk_cache_type)
+
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('storage_sources', 'dir_store')
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def insert(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor[ds.key(i)] = str(i) * 100
+    
+    def read_and_verify(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor.set_key(ds.key(i))
+            cursor.search()
+            self.assertEqual(cursor.get_value(), str(i) * 100)
+
+    def test_chunkcache04(self):
+        uris = ["table:chunkcache03", "table:chunkcache04"]
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in uris]
+
+        # Insert unpinned data into two tables. 
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+
+        # As we have not flushed yet, assert we have no newly inserted chunks. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted), 0)
+        
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+
+        # Assert that chunks are not pinned. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned), 0)
+
+        # Assert the new chunks are ingested. 
+        unpinned_ingest = self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted)
+        self.assertGreater(unpinned_ingest, 0)
+        
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in self.pinned_uris]
+
+        # Insert pinned data into two tables.
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+
+        # Assert the new chunks are ingested.
+        pinned_ingest = self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted)
+        self.assertGreater(pinned_ingest, unpinned_ingest)
```

Comment:
Why are we checking this?

Reply:
Not sure what you mean by 'this' but here we are checking here that ingest works for both pinned and unpinned chunks - and testing the possible paths. As the stat accumulates, we are checking the value against the previous amount and not zero.

Reply:
Sorry, I meant the following check - `self.assertGreater(pinned_ingest, unpinned_ingest)`

Why do we expect pinned_ingest to be greater than unpinned_ingest?

Reply:
Oh yeah the name can be confusing - we only expect it to be bigger not because of the pinned / unpinned status - but because we do the pinned_ingest after without reopening the connection so the statistic just adds up! I am just asserting that another set of ingests take place after calling ingest on another group of tables.

Reply:
ahh I see, to avoid the confusion, what do you think of the name `total_ingested_chunks`